include/TimingTools.h

```diff
@@ -98,6 +98,141 @@ namespace Mezzanine
+                TimeType GetIndexValueFromPercent(PreciseReal Percent) const;
+
+                /// @brief How many times was the timed item executed/
+                CountType Iterations = 0;
+                /// @brief How much was the total runtime with as much of the benchmark removed as possible.
+                TimeType Total = TimeType{0};
+                /// @brief The mean execution time; the Total time divided by the number of iterations.
+                TimeType Average = TimeType{0};
+                /// @brief The fastest (fewest time units) execution time. Defaults to one hour
+                TimeType Fastest = TimeType{0};
+                /// @brief The timing that beats out only 1 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan99Percent = TimeType{0};
+                /// @brief The timing that beats out 10 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan90Percent = TimeType{0};
+                /// @brief The meduan execution time; the execution time in the middle.
+                TimeType Median = TimeType{0};
+                /// @brief The timing that beats out 90 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan10Percent = TimeType{0};
+                /// @brief The timing that beats out 99 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan1Percent = TimeType{0};
+                /// @brief The slowest (most time units) execution time.
+                TimeType Slowest = TimeType{0};
+                /// @brief The raw times gathered by a test, sorted by performance.
+                TimingLists SortedTimings;
+                /// @brief The unsorted timings to help study caching effects.
+                TimingLists UnsortOriginalTimings;
+
+            };
+        RESTORE_WARNING_STATE
+
+        /// @brief Time a single execution of some functor
+        /// @tparam Functor Any function-like callable item which accepts no parameters and returns none.
+        /// @param  ToTime A functor to time the execution of.
+        /// @return A performanc profile as an instance of MicroBenchmarkResults.
+        template<typename Functor>
+        MicroBenchmarkResults MicroBenchmark(Functor&& ToTime)
+        {
+            TestTimer Bench;
+            ToTime();
+            return MicroBenchmarkResults{ {Bench.GetLength()} };
+        }
+
+        /// @brief Run the passed functor a number of times and track run times of these.
+        /// @tparam Functor Any function-like callable item which accepts no parameters and returns none.
+        /// @param ToTime A functor to time the execution of.
+        /// @return A performanc profile as an instance of MicroBenchmarkResults.
```

Comment:
performanc -> performance, I smell copy pasta.

Reply:
Let me go see what I did, whatever the root cause I blame to much coding after 3am.

Reply:
Yeah, there was some copypasta, but the output and inputs have a lot of similarities, just different algorithms.